tests/Serializer/AbstractSerializerTest.php

```diff
@@ -65,6 +65,41 @@ public function testObjectsAreNotStrings(): void
         $this->assertSame(['key' => 'value'], $result);
     }
 
+    public function traversableDataProvider(): \Generator
+    {
+        $getArray = static function (): array {
```

Comment:
What's the purpose of this anonymous function? I suggest to either inline the array in every place where it's used (easiest solution and also probably the cleanest given how little it is) or to declare it outside the function which to me looks as a pretty useless wrapper

Reply:
I had started with `$array = [ /* ... */ ];` (and use `$array` in place of `$getArray()` below) but then wanted to make sure to get a new copy each time (to avoid any risk of messing the internal pointer). I also wanted to avoid code duplication, but I will follow your suggestion and inline it :+1: